src/lib.rs

```diff
@@ -617,6 +618,22 @@ pub trait Itertools : Iterator {
         adaptors::step(self, n)
     }
 
+    /// Convert each item of the iterator using the `Into` trait.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use itertools::Itertools;
+    ///
+    /// (1i32..42i32).map_into().collect::<Vec<f64>>();
```

Comment:
Let's use an example with the turbofish maybe? Seems nice to put it like `.map_into::<f64>()` or is that ugly? :slightly_smiling_face:

Reply:
I'm not sure about that. In this context it would probably look odd (because I'd still have to use the turbofish on collect too, to tell it it's `Vec`).

Should I try to come up with some other example that makes more sense?

Reply:
Hm right, probably most users would avoid the double turbofish if they could! As you want, can be fixed later. Let's give it a go and see if we like it:

```rust
use itertools::Itertools;

(1i32..42i32).map_into::<f64>().collect::<Vec<_>>();
```

Or try a different example:

```rust
use itertools::Itertools;

let frequencies = [7, 5, 15, 5, 0, 2];
let average_value = frequencies.iter().cloned().map_into::<f64>().sum() / frequencies.len() as f64;
```

Uh, sometimes it just doesn't inspire me, the way we have to express these things (I think it has too much noise compared to the punch — `cloned()` is noise, and `iter()` is noise, both just to convert equivalentish things). Numeric applications really demand their own `.sum_as` or `.mean_value_as` which compute the accumulation using the specified type!

Reply:
I noticed itertools have `collect_vec`, so it can be used to avoid the double turbofish thing. Maybe that looks better?

Reply:
yep!